analyzer_test.go

```diff
@@ -63,3 +63,52 @@ func TestDeriveManifestAndLockDoesNotExist(t *testing.T) {
 		t.Fatalf("expected manifest & lock & err to be nil: m -> %#v l -> %#v err-> %#v", m, l, err)
 	}
 }
+
+func TestDeriveManifestAndLockCannotOpen(t *testing.T) {
+	dir, err := ioutil.TempDir("", "dep")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	fn := filepath.Join(dir, ManifestName)
+	fh, err := os.Create(fn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fh.Close()
+
+	err = os.Chmod(fn, 0222)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	a := analyzer{}
+
+	m, l, err := a.DeriveManifestAndLock(dir, "my/fake/project")
+	if m != nil || l != nil || err != nil {
+		t.Fatalf("expected manifest & lock & err to be nil: m -> %#v l -> %#v err-> %#v", m, l, err)
```

Comment:
Hmm...I looked, and this is indeed what we do right now. And I feel like it's bad. If the file's there, but we can't open it, then I think it's probably better to pass back the error.

Would you mind changing the return value over in `DeriveManifestAndLock()`, then change this test to suit?